desktop_server/src/com/mygdx/game/server/network/EndTurnService.java

```diff
@@ -0,0 +1,68 @@
+package com.mygdx.game.server.network;
+
+import com.artemis.World;
+import com.mygdx.game.core.ecs.component.PlayerToken;
+import com.mygdx.game.server.ecs.entityfactory.ComponentFactory;
+import com.mygdx.game.server.model.Client;
+import com.mygdx.game.server.model.GameRoom;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+import java.util.List;
+
+@Log
+public class EndTurnService {
+
+  private final int playerTokenComponentId;
+  private final PlayerToken playerToken;
+  private final GameRoom gameRoom;
+  private final GameRoomSyncer gameRoomSyncer;
+
+  @Inject
+  EndTurnService(
+      @NonNull World world,
+      @NonNull ComponentFactory componentFactory,
+      @NonNull GameRoom gameRoom,
+      @NonNull GameRoomSyncer gameRoomSyncer
+  ) {
+    this.playerTokenComponentId = componentFactory.createEntityId();
+    this.playerToken = world.getMapper(PlayerToken.class).create(this.playerTokenComponentId);
+    this.gameRoom = gameRoom;
+    this.gameRoomSyncer = gameRoomSyncer;
+  }
+
+  public void giveTurnToFirstPlayer() {
+    var clients = gameRoom.getClients().stream().toList();
+    log.info("First turn, start player " + clients.get(0).getPlayerUsername());
+    giveControlToPlayer(0, clients);
+  }
+
+  public void nextTurn(String playerNickname) {
+    log.info("Give control to player " + playerNickname);
+    var clients = gameRoom.getClients().stream().toList();
+    for (int i = 0; i < clients.size(); i++) {
+      if (clients.get(i).getPlayerUsername().equals(playerNickname)) {
+        giveControlToPlayer(i + 1, clients);
+      }
+    }
+  }
```

Comment:
czemu to ma być na podstawie nicków? nicki powinny móc się powtarzać

Reply:
a nie może być jakiegoś stanu, który zapisuje, który indeks gracza akurat teraz ma turę? to by wtedy cały ten serwis uprościło, takie coś może być zapisane w gameroom albo przez ECS i przepisanie tego na systemy

Reply:
no nie wiem, ja myslę że jest ok, gracze identyfikują się tokenem a nick to tylko wizualna właściwość, rozumiem że masz intuicję ponownego dołączania do gry ale narazie tego nie widzę